### ISSUE TITLE
BorderChildren not being generated in diagram

### ISSUE THREAD

**Issue by gruvy245** [awslabs/diagram-as-code#141]
## Issue detail
running against an example, it appears that the BorderChildren elements are not generated and link are not created.

```
❯ uname -a
Linux 25dfb146a6b5 5.15.153.1-microsoft-standard-WSL2 #1 SMP Fri Mar 29 23:14:13 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux

❯ awsdac --version
awsdac version dev

❯ go version
go version go1.21.13 linux/amd64
```
### command / error messages
```
$ awsdac alb-asg.yaml -o alb-asg.png -v
INFO[0000] input file path: alb-asg.yaml                
INFO[0000] Load DefinitionFiles section                 
INFO[0000] Fetch definition file from URL: https://raw.githubusercontent.com/awslabs/diagram-as-code/main/definitions/definition-for-aws-icons-light.yaml 
INFO[0000] [internal/cache/cache.go] FetchFile https://raw.githubusercontent.com/awslabs/diagram-as-code/main/definitions/definition-for-aws-icons-light.yaml 
INFO[0000] [internal/cache/cache.go] Found previous Etag cache. Use HTTP Etag value "7499768e9318b63663b26d7ce74d137039c0fa2489d079065c76fb5837229aad" 
INFO[0000] [internal/cache/cache.go] Server respond with HTTP 304 
INFO[0000] [internal/cache/cache.go] Use cache based on matched HTTP Etag 
INFO[0000] Read definition file from cache file: /home/vscode/.cache/awsdac/850960c5c172c277f880286d0349c39c-definition-for-aws-icons-light.yaml 
INFO[0000] [internal/cache/cache.go] FetchFile https://d1.awsstatic.com/webteam/architecture-icons/q2-2024/AWS-Architecture-Icon-Decks.pptx.d6671530ea426cd7ff7fb00423bccb120bfc69df.zip 
INFO[0000] [internal/cache/cache.go] Found previous Etag cache. Use HTTP Etag value "ccecb6dfad3955969e4ce76bd853af4a" 
INFO[0000] [internal/cache/cache.go] Server respond with HTTP 304 
INFO[0000] [internal/cache/cache.go] Use cache based on matched HTTP Etag 
INFO[0000] Load Resources section                       
INFO[0000] Load Resource: VPCPublicSubnet2Instance (AWS::EC2::Instance) 
INFO[0000] Load Resource: ALB (AWS::ElasticLoadBalancingV2::LoadBalancer) 
INFO[0000] Load Resource: IGW (AWS::EC2::InternetGateway) 
INFO[0000] Load Resource: Canvas (AWS::Diagram::Canvas) 
INFO[0000] Load Resource: AWSCloud (AWS::Diagram::Cloud) 
INFO[0000] Load Resource: VPCPublicSubnet1 (AWS::EC2::Subnet) 
INFO[0000] Load Resource: VPCPublicSubnet1Instance (AWS::EC2::Instance) 
INFO[0000] Load Resource: VPC (AWS::EC2::VPC)           
INFO[0000] Load Resource: VPCPublicStack (AWS::Diagram::HorizontalStack) 
INFO[0000] Load Resource: VPCPublicSubnet2 (AWS::EC2::Subnet) 
INFO[0000] Load Resource: User (AWS::Diagram::Resource) 
INFO[0000] Associate children with parent resources     
INFO[0000] Add child(VPCPublicStack) on VPC             
INFO[0000] Add child(ALB) on VPC                        
INFO[0000] Add child(VPCPublicSubnet1) on VPCPublicStack 
INFO[0000] Add child(VPCPublicSubnet2) on VPCPublicStack 
INFO[0000] Add child(VPCPublicSubnet2Instance) on VPCPublicSubnet2 
INFO[0000] Add child(AWSCloud) on Canvas                
INFO[0000] Add child(User) on Canvas                    
INFO[0000] Add child(VPC) on AWSCloud                   
INFO[0000] Add child(VPCPublicSubnet1Instance) on VPCPublicSubnet1 
INFO[0000] Add Links section                            
INFO[0000] Add link(ALB-VPCPublicSubnet1Instance)       
INFO[0000] Add link(ALB-VPCPublicSubnet2Instance)       
INFO[0000] Add link(IGW-ALB)                            
INFO[0000] Add link(User-IGW)                           
INFO[0000] --- Draw diagram ---                         
INFO[0000] Scale                                        
INFO[0000] Scale AWS Cloud                              
INFO[0000] Scale VPC                                    
INFO[0000] Scale                                        
INFO[0000] Scale Public Subnet                          
INFO[0000] Scale Instance                               
INFO[0000] Scale Public Subnet                          
INFO[0000] Scale Instance                               
INFO[0000] Scale Application Load Balancer              
INFO[0000] Scale User                                   
INFO[0000] Link Drawing                                 
INFO[0000] arrowHead.Length:"10", arrowHead.Width:""    
INFO[0000] arrowHead.Length:"10", arrowHead.Width:""    
INFO[0000] Open Arrow Head drawing                      
INFO[0000] Link Drawing                                 
INFO[0000] arrowHead.Length:"10", arrowHead.Width:""    
INFO[0000] arrowHead.Length:"10", arrowHead.Width:""    
INFO[0000] Open Arrow Head drawing                      
INFO[0000] Save alb-asg.png                             
[Completed] AWS infrastructure diagram generated: alb-asg.png
```

### Yaml template (From examples/alb-asg.yaml)
```
Diagram:
  DefinitionFiles:
    - Type: URL
      Url: "https://raw.githubusercontent.com/awslabs/diagram-as-code/main/definitions/definition-for-aws-icons-light.yaml"
    #- Type: LocalFile
    #  LocalFile: ./definitions/definition-for-aws-icons-light.yaml

  Resources:
    Canvas:
      Type: AWS::Diagram::Canvas
      Direction: vertical
      Children:
        - AWSCloud
        - User
    AWSCloud:
      Type: AWS::Diagram::Cloud
      Direction: vertical
      Preset: AWSCloudNoLogo
      Align: center
      Children:
        - VPC
    # VPC
    VPC:
      Type: AWS::EC2::VPC
      Direction: vertical
      Children:
        - VPCPublicStack
        - ALB
      BorderChildren:
        - Position: S
          Resource: IGW
    VPCPublicStack:
      Type: AWS::Diagram::HorizontalStack
      Children:
        - VPCPublicSubnet1
        - VPCPublicSubnet2
    VPCPublicSubnet1:
      Type: AWS::EC2::Subnet
      Preset: PublicSubnet
      Children:
        - VPCPublicSubnet1Instance
    VPCPublicSubnet1Instance:
      Type: AWS::EC2::Instance
    VPCPublicSubnet2:
      Type: AWS::EC2::Subnet
      Preset: PublicSubnet
      Children:
        - VPCPublicSubnet2Instance
    VPCPublicSubnet2Instance:
      Type: AWS::EC2::Instance
    ALB:
      Type: AWS::ElasticLoadBalancingV2::LoadBalancer
      Preset: Application Load Balancer
    IGW:
      Type: AWS::EC2::InternetGateway
    User:
      Type: AWS::Diagram::Resource
      Preset: User

  Links:
    - Source: ALB
      SourcePosition: NNW
      Target: VPCPublicSubnet1Instance
      TargetPosition: SSE
      TargetArrowHead:
        Type: Open
    - Source: ALB
      SourcePosition: NNE
      Target: VPCPublicSubnet2Instance
      TargetPosition: SSW
      TargetArrowHead:
        Type: Open
    - Source: IGW
      SourcePosition: N
      Target: ALB
      TargetPosition: S
      TargetArrowHead:
        Type: Open
    - Source: User
      SourcePosition: N
      Target: IGW
      TargetPosition: S
      TargetArrowHead:
        Type: Open
...
```
![image](https://github.com/user-attachments/assets/b98cc253-3ef6-42ff-82e3-69470ff20e4d)

**Comment by ugwis** [comment]
@gruvy245 Thank you for reporting the issue!
It was not included in v0.21.6, so we just released v0.21.7. Could you please try reinstalling latest to see if it improves?

**Comment by gruvy245** [comment]
> @gruvy245 Thank you for reporting the issue! It was not included in v0.21.6, so we just released v0.21.7. Could you please try reinstalling the latest to see if it improves?

Looks good now.  Thanks!
